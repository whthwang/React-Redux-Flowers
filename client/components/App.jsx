import React, { Component } from 'react';
import { connect } from 'react-redux';
import Users from './Users.jsx';
import UserDetails from './UserDetails.jsx';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleSelect = this.handleSelect.bind(this);
  }

  componentDidMount() {
    this.props.fetchAll();
  }

  handleSelect(e) {
    let { selectUser, allUsers } = this.props;
    let person = allUsers[e.target.id];
    selectUser(person);
  }

  render() {
    return (
      <div className='app'>
        <div>
          <Users allUsers={this.props.allUsers} handleSelect={this.handleSelect} />
        </div>
        <div className='user-details'>
          <UserDetails user={this.props.selectedUser}/>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    allUsers: state.allUsers,
    selectedUser: state.selectedUser
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchAll: () => dispatch({ type: 'FETCH_USERS' }),
    selectUser: (person) => dispatch({ type: 'SELECT_USER', payload: person })
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(App);