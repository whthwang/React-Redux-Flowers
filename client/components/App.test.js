import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from './App.jsx';

describe('testing components', () => {
  it('expect to render App component', () => {
    expect(1).toEqual(1);
  })
})