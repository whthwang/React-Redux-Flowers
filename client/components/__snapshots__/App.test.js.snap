// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for rendered components expect UserDetails component to match snapshot 1`] = `ShallowWrapper {}`;

exports[`Test for rendered components expect Users component to render 1`] = `1`;

exports[`Tests for snapshots expect UserDetails component to match snapshot 1`] = `ShallowWrapper {}`;

exports[`Tests for snapshots expect Users component to match snapshot 1`] = `ShallowWrapper {}`;
